src/vsphere_cpi/spec/unit/cloud/vsphere/drs_rules/vm_group_spec.rb

```diff
@@ -107,4 +110,47 @@ def with_lock
       end
     end
   end
+
+  describe '#get_delete_rule_spec' do
+    before do
+      allow(datacenter_cluster).to receive(:configuration_ex).and_return(configuration_ex)
+    end
+    context 'when no rule exists' do
+      before do
+        allow(configuration_ex).to receive(:rule).and_return(nil)
+      end
+      it 'returns nil' do
+        expect(vm_group.send(:get_delete_rule_spec, *[vm_groups])).to be(nil)
+      end
+    end
+    context 'when vm_groups passed are empty' do
+      before do
+        allow(configuration_ex).to receive(:rule).and_return(['fake-rule'])
+      end
+      it 'returns nil' do
+        expect(vm_group.send(:get_delete_rule_spec, *[nil])).to be(nil)
```

Comment:
spec says empty vm_groups but you are passing a nil here

Reply:
Corrected.